File: php/src/Model/TaskDataSummary.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Counts of the rich data hanging off a task; populated on single-task reads only.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class TaskDataSummary
{
    public function __construct(
        public readonly bool $hasContext,
        public readonly int $referenceCount,
        public readonly int $attachmentCount,
        public readonly int $commentCount,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            hasContext: (bool) ($data['hasContext'] ?? false),
            referenceCount: (int) ($data['referenceCount'] ?? 0),
            attachmentCount: (int) ($data['attachmentCount'] ?? 0),
            commentCount: (int) ($data['commentCount'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/TaskDataSummary.java
package io.fivexer.sdk.model;

/**
 * Counts of the rich data hanging off a task; populated on single-task reads only.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class TaskDataSummary {
    private boolean hasContext;
    private int referenceCount;
    private int attachmentCount;
    private int commentCount;

    public boolean isHasContext() { return hasContext; }
    public int getReferenceCount() { return referenceCount; }
    public int getAttachmentCount() { return attachmentCount; }
    public int getCommentCount() { return commentCount; }
}
